### PR TITLE
asynchronously fetch token data

### LINKED ISSUE #868: Await Token Info in Parallel
This method is currently inefficient and slow when used to fetch many tokens at once. This could be easily improved by sending out EVM queries in parallel (i.e. with `await Promise.all`).

https://github.com/gnosis/dex-contracts/blob/f5a3dbc696c12bb6af2ea5eee97c703084c84d1a/src/fee_token_liquidity.ts#L29-L51

While improving this, it would also be nice to refactor this to use the batch exchange viewer contract changes that were introduced in #860

### PR DESCRIPTION
**Pull request by bh2smith**
The previous implementation was naive and this one is much much faster!

closes #868  (Note that it was suggested to use web3-BatchRequest, but this is efficient as is and the web3 option seems to require passing around additional artifacts).

**Test Plan**
Can't really be tested without a wrapper script. The only script using this within the project is `scripts/ensure_owl_liquidity.ts`

### CODE REVIEW

File: src/fee_token_liquidity.ts [hunk 1]
```diff
@@ -26,30 +26,32 @@ export async function fetchTokenInfoFromExchange(
...
+      try {
+        const tokenInstance = await ERC20.at(tokenAddress);
+        const symbol = await tokenInstance.symbol();
+        const decimals = (await tokenInstance.decimals()).toNumber();
+        tokenInfo = {
+          id: id,
+          symbol: symbol,
+          decimals: decimals,
+          address: tokenAddress,
+        };
```
Comment (fedgiac) [inline src/fee_token_liquidity.ts]: Another small efficiency improvement could be this:
```suggestion
      const [symbol, decimals] = await Promise.all([
        tokenInstance.symbol(),
        tokenInstance.decimals(),
      ]);
      tokenInfo = {
        id: id,
        symbol: symbol,
        decimals: decimals.toNumber(),
        address: tokenAddress,
      };
```

**Review by fedgiac — APPROVED** [review]
Looks good!

This would be a perfect candidate to test out batch web3 requests at some point.

**Comment by bh2smith** [comment]
> This would be a perfect candidate to test out batch web3 requests at some point.

Yes! I had forgotten. Maybe I should do that now!

**Review by fleupold — APPROVED** [review]
> The previous implementation was naive and this one is much much faster!

Would be awesome to quantify how much faster - maybe just using the unix `time` tool before and after to get an idea.

**Comment by bh2smith** [comment]
@fleupold 

> This would be a perfect candidate to test out batch web3 requests at some point.

**Old Script:**
    Fetching info for 18 tokens
    Execution Time: 66.557 seconds

**New Script:** 
    Fetching info for 18 tokens
    Execution Time: 3.478 seconds

Appears to be about 22x faster.


This was executed with the following script:

```ts
import { fetchTokenInfoFromExchange } from "../src/fee_token_liquidity";
import { getBatchExchange } from "./util";

module.exports = async (callback: Truffle.ScriptCallback) => {
  try {
    const exchange = await getBatchExchange(artifacts);
    const numTokens = (await exchange.numTokens()).toNumber();
    const tokens = Array(numTokens)
      .fill(0)
      .map((_, i) => i);
    console.log(`Fetching info for ${numTokens} tokens`);
    console.time("Execution Time");
    await fetchTokenInfoFromExchange(exchange, tokens, artifacts);
    console.timeEnd("Execution Time");
    callback();
  } catch (error) {
    callback(error);
  }
};
```

using 

```sh
npx truffle exec build/common/scripts/test_script.js --network rinkeby
```

**Comment by fleupold** [comment]
Awesome speed improvement!

**Pull request merged**